batcher/aligned-batcher/src/lib.rs

```diff
@@ -832,29 +731,44 @@ impl Batcher {
 
         // Set the batch posting flag to true
         *batch_posting = true;
-
-        let batch_queue_copy = batch_state.batch_queue.clone();
-        match batch_queue::try_build_batch(batch_queue_copy, gas_price, self.max_batch_size) {
-            Ok((resulting_batch_queue, finalized_batch)) => {
-                // Set the batch queue to batch queue copy
-                batch_state.batch_queue = resulting_batch_queue;
-                batch_state.update_user_proofs_in_batch_and_min_fee();
-                Some(finalized_batch)
-            }
-            Err(BatcherError::BatchCostTooHigh) => {
-                // We cant post a batch since users are not willing to pay the needed fee, wait for more proofs
-                info!("No working batch found. Waiting for more proofs...");
-                *batch_posting = false;
-                None
-            }
-            // FIXME: We should refactor this code and instead of returning None, return an error.
-            // See issue https://github.com/yetanotherco/aligned_layer/issues/1046.
-            Err(e) => {
-                error!("Unexpected error: {:?}", e);
-                *batch_posting = false;
-                None
-            }
+        let batch_queue_copy = batch_state_lock.batch_queue.clone();
+        let (resulting_batch_queue, finalized_batch) =
+            batch_queue::try_build_batch(batch_queue_copy, gas_price, self.max_batch_size)
+                .inspect_err(|e| {
+                    *batch_posting = false;
+                    match e {
+                        // We can't post a batch since users are not willing to pay the needed fee, wait for more proofs
+                        BatcherError::BatchCostTooHigh => {
+                            info!("No working batch found. Waiting for more proofs")
+                        }
+                        // FIXME: We should refactor this code and instead of returning None, return an error.
+                        // See issue https://github.com/yetanotherco/aligned_layer/issues/1046.
+                        e => error!("Unexpected error: {:?}", e),
+                    }
+                })
+                .ok()?;
+
+        batch_state_lock.batch_queue = resulting_batch_queue;
+        let updated_user_proof_count_and_min_fee =
+            batch_state_lock.get_user_proofs_in_batch_and_min_fee();
+
+        let user_addresses: Vec<Address> = batch_state_lock.user_states.keys().cloned().collect();
```

Comment:
I believe you're only iterating the addresses once. The `collect` is unnecessary.